File: src/main/java/com/xgarage/app/dto/RequestDto.java
package com.xgarage.app.dto;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@NoArgsConstructor
public class RequestDto {
    private Long id;
    private Long status;
    private Timestamp submissionDate;
    private Long userId;
    private String firstName;
    private String requestTitle;
    private long submittedBids ;
    private long rejectedBids;
    private BidDto selectedBid;

    private Long jobId;

    public RequestDto(Long id, Long status, Timestamp submissionDate, Long userId, String firstName, String requestTitle, long submittedBids, long rejectedBids, BidDto selectedBid, Long claimId) {
        this.id = id;
        this.status = status;
        this.submissionDate = submissionDate;
        this.userId = userId;
        this.firstName = firstName;
        this.requestTitle = requestTitle;
        this.submittedBids = submittedBids;
        this.rejectedBids = rejectedBids;
        this.selectedBid = selectedBid;
        this.jobId = claimId;
    }
}

File: src/app/xgarage/core/dto/requestdto.ts
export interface RequestDto{
    id?: number;
    status?: number;
    submissionDate?: Date;
    userId?: number;
    privacy?: string;
    firstName?: string;
    requestTitle?: string;
    submittedBids?: number;
    rejectedBids?: number;
    jobId?: number;
    claimId?: number;
    jobNo?: string;
    claimNo?: string;
    qtty?: number;
}